pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -535,6 +563,13 @@ func (r *ReconcileBareMetalHost) actionProvisioning(prov provisioner.Provisioner
 	}
 	info.log.Info("provisioning")
 
+	if clearRebootAnnotations(info.host) {
+		if err := r.client.Update(context.TODO(), info.host); err != nil {
+			return actionError{errors.Wrap(err, "failed to remove reboot annotations from host")}
+		}
+		return actionContinueNoWrite{}
+	}
```

Comment:
I forgot to mention, there is a slight race here in that we will continue to clear reboot annotations up until the provisioning is complete. So if the server completes booting, starts up, and needs remediation before Ironic notices that provisioning is complete, the reboot annotations will be cleared without actually rebooting.
In practice this doesn't seem like a plausible scenario, and it's difficult to know what to do about it, so I'm not too worried.